ui_components/src/village/Village.js

```diff
@@ -0,0 +1,316 @@
+﻿import { apiFetch } from "../utils/network.js";
+
+function Village({
+    playerSeat,
+    villageName,
+    villageAPI,
+    policyData,
+    populationData,
+    seatData,
+    pointsData,
+    diplomacyData,
+    resourceData,
+    clanData,
+}) {
+    const [seatDataState, setSeatDataState] = React.useState(seatData);
+    const [resourceDataState, setResourceDataState] = React.useState(resourceData);
+    const [playerSeatState, setPlayerSeatState] = React.useState(playerSeat);
+    const [modalState, setModalState] = React.useState("closed");
+    const resourceDays = React.useRef(1);
+    const resourceDaysDisplay = React.useRef("daily");
```

Comment:
Curious, why refs here instead of state?

Reply:
Yeah we can just do state, will change that.